ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/sparse_encoding/SparseEncodingTranslator.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.engine.algorithms.sparse_encoding;
+
+import ai.djl.huggingface.tokenizers.Encoding;
+import ai.djl.huggingface.tokenizers.HuggingFaceTokenizer;
+import ai.djl.modality.Input;
+import ai.djl.modality.Output;
+import ai.djl.ndarray.NDArray;
+import ai.djl.ndarray.NDList;
+import ai.djl.ndarray.NDManager;
+import ai.djl.ndarray.types.DataType;
+import ai.djl.translate.Batchifier;
+import ai.djl.translate.ServingTranslator;
+import ai.djl.translate.TranslatorContext;
+import org.opensearch.ml.common.output.model.MLResultDataType;
+import org.opensearch.ml.common.output.model.ModelTensor;
+import org.opensearch.ml.common.output.model.ModelTensors;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.file.Path;
+import java.util.*;
+
+import static org.opensearch.ml.common.CommonValue.ML_MAP_RESPONSE_KEY;
+
+public class SparseEncodingTranslator implements ServingTranslator {
```

Comment:
Looks like the SparseEncodingTranslator has same tokenizer and methods like getBatchifier, prepare, processInput, setArguments as the SentenceTransformerTextEmbeddingTranslator. Only different function is ProcessOutput. So it's better to have this translator inherited from SentenceTransformerTextEmbeddingTranslator.

Reply:
Already extract an abstract class called SentenceTransformerTranslator and both dense/sparse sentence translator will extend this one.